plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -640,19 +640,80 @@ public function get_total_shipping_refunded( $order ) {
 		return 0;
 	}
 
+	/**
+	 * Finds an Order ID based on an order key.
+	 *
+	 * @param string $order_key An order key has generated by.
+	 * @return int The ID of an order, or 0 if the order could not be found
+	 */
 	public function get_order_id_by_order_key( $order_key ) {
-		// TODO: Implement get_order_id_by_order_key() method.
-		return 0;
+		global $wpdb;
+
+		$orders_table = self::get_orders_table_name();
+		$op_table     = self::get_operational_data_table_name();
+
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		return $wpdb->get_var(
```

Comment:
What do you think of:

```suggestion
		return (int) $wpdb->get_var(
```

Reason being the result of a call to `$wpdb->get_var()` is always a string (so an integer result will actually be something like `(string) '123'`) or else it will be `null`. This would ensure the result matches the type given in the `@return` tag.